part1/src/App2.js

```diff
@@ -1,30 +1,74 @@
 import { useState } from 'react'
 
+const Button = (props) => {
+  const type = props.type
+  const name = props.name
+  const setState = props.setState
+
+  return (
+    <button onClick={() => setState(type + 1) }> {name} </button>
+  )
+}
+
+const StatisticLine = (props) => {
+  const text = props.text
+  const value = props.value
+
+  return (
+    <tr>
+      <td>{text}</td>
+      <td>{value}</td>
+    </tr>
+  )
+}
+
+const Statistics = (props) => {
+  const all = props.good + props.bad + props.neutral
+  const average = (props.good - props.bad) / all
+  const positive = props.good / all * 100
+  const good = props.good
+  const neutral = props.neutral
+  const bad = props.bad
```

Comment:
maybe use the variables you defined for `props.good` , `props.neutral` and `props.bad` for all, average and positive?